src/react.tsx

```diff
@@ -1,21 +1,37 @@
 import { DocumentNode } from "graphql"
+import { renderToStaticMarkup } from "react-dom/server"
 
 import { Query, FetchPolicy } from "./Query"
 import { MSTGQLStore } from "./MSTGQLStore"
 
 // import react namespace only; statement gets removed after transpiling
-declare var ReactNamespace: typeof import('react')
+declare var ReactNamespace: typeof import("react")
 
 export type QueryLike<STORE, DATA> =
   | ((store: STORE) => Query<DATA>)
   | Query<DATA>
   | string
   | DocumentNode
 
-export function createStoreContext<STORE extends typeof MSTGQLStore.Type>(React: typeof ReactNamespace) {
+export function createStoreContext<STORE extends typeof MSTGQLStore.Type>(
```

Comment:
Perhaps I am mistaken, but it could be more correct like this and you wouldn't need to explicitly type those `__promises` later.
```suggestion
export function createStoreContext<STORE extends Instance<typeof MSTGQLStore>(
```
Note: Instance is imported from MST.

Or it might be better to export type alias from `MSTGQLStore.ts` directly.

Reply:
@FredyC you will have to ask @mweststrate about that, that is how it is typed elsewhere

Reply:
I know, but can you try it? I believe it should work.

Reply:
It's both the same thing, (I actually don't know what is idiomatic, cc @xaviergonz )

Reply:
I was mainly concerned why is it necessary to explicitly type those `__promises`. With the correct `STORE` type, it should be able to infer it, shouldn't it?